### ISSUE TITLE
Request for the feature to get the size of ndarray in Taichi kernel.

### ISSUE THREAD

**Issue by daichi-ishida** [taichi-dev/taichi#6799]
In numpy ndarray, the number of elements can be obtained with `ndarray.size`
but in Taichi, it raise an AttributeError exception

Passing the numpy size to the kernel would solve the problem, but it's not "Zen of Python" style to pass it with ndarray just for that purpose, and I thought it would be useful to get it in the kernel.

**Comment by ailzhang** [comment]
@daichi-ishida just to double check, are you looking for a way to use ndarray shape in ti.kernel like this: https://github.com/taichi-dev/taichi/blob/master/tests/cpp/aot/python_scripts/mpm88_graph_aot.py#L63?

**Comment by daichi-ishida** [comment]
@ailzhang No, as you pointed out, when the dimension of ndarray is small and known, it can be calculated by explicitly multiplying each element of `ndarray.shape`.
But, since Taichi does not provide `ndim` either, I think there is no way to get the size of an arbitrary-dimension ndarray currently.

**Comment by ailzhang** [comment]
oh I see so you were asking for number of all elements in ndarray. Yea that's a valid feature request for convenience.
